extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -381,17 +412,59 @@ public String toString()
     ENTRY_CLOSED
   }
 
-  public final class VersionedCache implements CacheState, AutoCloseable
+  /**
+   * This builder class holds a {@link VersionedCache} object,
+   * For handling failures while loading cache, it exposes close() method which will close {@link CacheHandler}
+   */
+  public final class VersionedCacheBuilder
```

Comment:
Why is this class not a `Closeable`? You can use the `try-resource` clause instead of [doing this](https://github.com/apache/druid/pull/8272/files#diff-4d4be3477c4c0924f89ef01c6a7362adR103).